### ISSUE TITLE
Suggestions for project enhancements

### ISSUE THREAD

**Issue by ThomasVitale** [diagridio/dapr-spring-boot-starter#22]
The starter dependency provided by this project is quite convenient for setting up a local development workflow based on Testcontainers, thank you for that! 

I'd like to share some suggestions to adopt a more modular strategy to offer different dependencies for different use cases, also distinguishing between compile-scope and test-scope. The strategy is commonly adopted for Spring Boot extensions (as an example, here's the [LangChain4j Spring Boot ](https://github.com/ThomasVitale/langchain4j-spring-boot) extension I've been working on recently).

A possible project structure could be the following:

```
dapr-spring-boot
|--- dapr-spring-boot-autoconfigure
|--- dap-spring-boot-starters
      |--- dapr-client-spring-boot-starter
      |--- dapr-actors-spring-boot-starter
      |--- dapr-workflows-spring-boot-starter
|--- dapr-spring-boot-docker-compose
|--- dapr-spring-boot-testcontainers
```

- The `dapr-spring-boot-autoconfigure` module would contain autoconfiguration for the different components provided by the Dapr Java SDK libraries, with configuration properties for customizations, and conditionals to enable certain features selectively (for example, the actors and workflows support might be enabled conditionally). It could also include the necessary configuration to make Dapr GraalVM-ready and allow compiling applications using Dapr to native executables. This module would deprecate the old `sdk-springboot` module in the Java SDK project (the one used for Spring Boot 2).
- There could a few starter dependencies with focused use cases, with the `dapr-client-spring-boot-starter` being the primary one and providing all the basic features to use Dapr. Among the other things, a `DaprClient` would be auto configured and ready for auto wiring.
- The `dapr-spring-boot-docker-compose` module would support Dapr dev services based on Docker Compose. This would be used as a "development only" dependency in application projects, ensuring no extra dependency is leaked into the final production artifacts.
- The `dapr-spring-boot-testcontainers` module would support Dapr dev services based on Testcontainers. This would be used as a "test and development only" dependency in application projects, ensuring no extra dependency is leaked into the final production artifacts.

An extra module could be `dapr-spring-cloud-bindings`, providing support for the Kubernetes Service Binding Specification and allowing automatic configuration of the Dapr client when running on Kubernetes.

What do you think? I'm available to help with this if there's interest.

**Comment by salaboy** [comment]
Yes this is great.. I think we should refactor what we have to follow that approach.. 

Something that you mentioned in a PR is the dependency with Spring Boot 2.x in the spring-boot-starter.. but we also have the 3.x version available.. so maybe we can cover both in separate modules? https://github.com/dapr/java-sdk/tree/master/sdk-springboot3

Any recommendations are more than welcome.

**Comment by salaboy** [comment]
the new structure on the `new-spring-boot` branch reflects this structure
